From 82489b51443e1280dfb9fb251ea2693df1809aec Mon Sep 17 00:00:00 2001
From: Pascal Terjan <pterjan@mandriva.com>
Date: Mon, 3 May 2010 14:01:22 +0200
Subject: [PATCH] Force GduPresentable ids to be UTF-8 (#616198)

GduPresentable created in gdu pool include intheir id some strings
in local encoding like _("Peripheral Devices"). This patch
enforces them to be UTF-8.

This fixes a crash of gvfs-gdu-volume-monitor when USB devices are
available on a non UTF-8 system.
---
 src/gdu/gdu-pool.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/src/gdu/gdu-pool.c b/src/gdu/gdu-pool.c
index bd5eccf..cf6be53 100644
--- src/gdu/gdu-pool.c.orig
+++ src/gdu/gdu-pool.c
@@ -473,6 +473,9 @@ gdu_pool_class_init (GduPoolClass *klass)
                               g_cclosure_marshal_VOID__OBJECT,
                               G_TYPE_NONE, 1,
                               GDU_TYPE_PRESENTABLE);
+#ifdef HAVE_BIND_TEXTDOMAIN_CODESET
+	bind_textdomain_codeset (GETTEXT_PACKAGE, "UTF-8");
+#endif
 }
 
 static void
-- 
1.7.1

